simapp/cmd/simd/main.go

```diff
@@ -1,118 +1,37 @@
 package main
 
 import (
-	"encoding/json"
-	"io"
-
 	"github.com/spf13/cobra"
-	"github.com/spf13/viper"
-	abci "github.com/tendermint/tendermint/abci/types"
 	"github.com/tendermint/tendermint/libs/cli"
-	"github.com/tendermint/tendermint/libs/log"
-	tmtypes "github.com/tendermint/tendermint/types"
-	dbm "github.com/tendermint/tm-db"
 
-	"github.com/cosmos/cosmos-sdk/baseapp"
-	"github.com/cosmos/cosmos-sdk/client/debug"
-	"github.com/cosmos/cosmos-sdk/client/flags"
-	"github.com/cosmos/cosmos-sdk/server"
 	"github.com/cosmos/cosmos-sdk/simapp"
-	"github.com/cosmos/cosmos-sdk/store"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
-	genutilcli "github.com/cosmos/cosmos-sdk/x/genutil/client/cli"
-	"github.com/cosmos/cosmos-sdk/x/staking"
 )
 
-const flagInvCheckPeriod = "inv-check-period"
-
-var invCheckPeriod uint
-
 func main() {
-	appCodec, cdc := simapp.MakeCodecs()
 
+	cobra.EnableCommandSorting = false
+
+	// Read in the configuration file for the sdk
 	config := sdk.GetConfig()
 	config.SetBech32PrefixForAccount(sdk.Bech32PrefixAccAddr, sdk.Bech32PrefixAccPub)
 	config.SetBech32PrefixForValidator(sdk.Bech32PrefixValAddr, sdk.Bech32PrefixValPub)
 	config.SetBech32PrefixForConsensusNode(sdk.Bech32PrefixConsAddr, sdk.Bech32PrefixConsPub)
 	config.Seal()
 
-	ctx := server.NewDefaultContext()
-	cobra.EnableCommandSorting = false
-	rootCmd := &cobra.Command{
-		Use:               "simd",
-		Short:             "Simulation Daemon (server)",
-		PersistentPreRunE: server.PersistentPreRunEFn(ctx),
+	rootComd := &cobra.Command{
+		Use:   "simd",
+		Short: "simulation app binary",
 	}
 
-	rootCmd.AddCommand(
-		genutilcli.InitCmd(ctx, cdc, simapp.ModuleBasics, simapp.DefaultNodeHome),
-		genutilcli.CollectGenTxsCmd(ctx, cdc, banktypes.GenesisBalancesIterator{}, simapp.DefaultNodeHome),
-		genutilcli.MigrateGenesisCmd(ctx, cdc),
-		genutilcli.GenTxCmd(
-			ctx, cdc, simapp.ModuleBasics, staking.AppModuleBasic{},
-			banktypes.GenesisBalancesIterator{}, simapp.DefaultNodeHome, simapp.DefaultCLIHome,
-		),
-		genutilcli.ValidateGenesisCmd(ctx, cdc, simapp.ModuleBasics),
-		AddGenesisAccountCmd(ctx, cdc, appCodec, simapp.DefaultNodeHome, simapp.DefaultCLIHome),
-		flags.NewCompletionCmd(rootCmd, true),
-		testnetCmd(ctx, cdc, simapp.ModuleBasics, banktypes.GenesisBalancesIterator{}),
-		debug.Cmd(cdc))
+	// Construct Root Command
+	rootComd = addClientCommands(rootComd)
+	rootComd = addDaemonCommands(rootComd)
 
-	server.AddCommands(ctx, cdc, rootCmd, newApp, exportAppStateAndTMValidators)
-
-	// prepare and add flags
-	executor := cli.PrepareBaseCmd(rootCmd, "GA", simapp.DefaultNodeHome)
```

Comment:
where did this guy go? is it not needed?

Reply:
No, it is needed but It is already indirectly called in

```
executor := cli.PrepareMainCmd(rootComd, "GA", simapp.DefaultNodeHome)
```
If we added it explicitly again, It will be called twice and the cobra doesn't like it.